base/weakkeydict.jl

```diff
@@ -80,7 +80,7 @@ function _cleanup_locked(h::WeakKeyDict)
     return h
 end
 
-sizehint!(d::WeakKeyDict, newsz) = sizehint!(d.ht, newsz)
+sizehint!(d::WeakKeyDict, newsz; shrink = true) = sizehint!(d.ht, newsz; shrink = shrink)
```

Comment:
Should we lock the dictionary here, as for all the other operations?

Reply:
yeah, seems like this should have a lock around it too
```suggestion
sizehint!(d::WeakKeyDict, newsz; shrink = true) = @lock d sizehint!(d.ht, newsz; shrink = shrink)
```